tchannel/testing/vcr/cassette.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+from itertools import chain
+from collections import deque
+
+from . import types
+from . import yaml
+from .exceptions import RequestNotFoundError, UnsupportedVersionError
+
+
+# Version of the storage format.
+VERSION = 1
+
+
+class Cassette(object):
+    """Represents a series of recorded interactions."""
+
+    def __init__(self, path):
+        """Initialize a new cassette.
+
+        :param path:
+            File path at which this cassette will be stored.
+        """
+        self.path = path
+
+        # Number of times interactions from this cassette have been played.
+        self.play_count = 0
+
+        self._available = deque()
+        self._played = deque()
+        self._recorded = deque()
+
+        self._load()
+
+    @property
+    def data(self):
+        """Get all known data for this cassette."""
+
+        return deque(
+            chain(self._played, self._available, self._recorded)
+        )
+
+    def _load(self):
+        try:
+            with open(self.path, 'r') as f:
```

Comment:
That's true. Any way that's the generated file. For example if you change the test file and make a change to our request, response objects as well, They old data in the file may no longer useful for the new request, response. If at the moment, you run the test, test will broken because the vcr will try to load from old yaml file. That's why we should have a command similar as 'make clean' in the Makefile so that we can clean the old vcr file.